spec/javascripts/components/global-banner-spec.js

```diff
@@ -0,0 +1,200 @@
+
+      for (var i = 0; i < cookies.length; i++) {
+        var cookie = cookies[i]
+        var eqPos = cookie.indexOf('=')
+        var name = eqPos > -1 ? cookie.substr(0, eqPos) : cookie
+        document.cookie = name + '=;expires=Thu, 01 Jan 1970 00:00:00 GMT'
+      }
+    }
+
+    beforeEach(function () {
+      element = $(
+        '<div id="global-banner" data-module="global-banner" data-banner-version="5">' +
+          '<a href="/register-to-vote" class="govuk-link js-call-to-action">Register to Vote</a>' +
+        '</div>'
+      )
+      $('html').append(element)
+      $('html').removeClass('show-global-banner')
+      deleteAllCookies()
+      window.GOVUK.setConsentCookie({ settings: true })
+    })
+
+    it('sets global_banner_seen cookie', function () {
+      globalBanner = new GOVUK.Modules.GlobalBanner(element[0])
+      globalBanner.init()
+
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).count).toBe(0)
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).version).toBe(5)
+      expectGlobalBannerToShow()
+      expectGa4AttributeToExist()
+    })
+
+    it('sets cookie to default value if current cookie is old (prior to versioning mechanism)', function () {
+      GOVUK.setCookie('global_banner_seen', 1)
+      globalBanner = new GOVUK.Modules.GlobalBanner(element[0])
+      globalBanner.init()
+
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).count).toBe(0)
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).version).toBe(5)
+
+      expectGlobalBannerToShow()
+      expectGa4AttributeToExist()
+    })
+
+    it('resets cookie if version number is out of date, if count below 3', function () {
+      GOVUK.setCookie('global_banner_seen', JSON.stringify({ count: 1, version: 1 }))
+      globalBanner = new GOVUK.Modules.GlobalBanner(element[0])
+      globalBanner.init()
+
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).count).toBe(0)
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).version).toBe(5)
+      expectGlobalBannerToShow()
+      expectGa4AttributeToExist()
+    })
+
+    it('resets cookie if version number is out of date, if count above 3', function () {
+      GOVUK.setCookie('global_banner_seen', JSON.stringify({ count: 10, version: 1 }))
+      globalBanner = new GOVUK.Modules.GlobalBanner(element[0])
+      globalBanner.init()
+
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).count).toBe(0)
+      expect(parseCookie(GOVUK.getCookie('global_banner_seen')).version).toBe(5)
+      expectGlobalBannerToShow()
+      expectGa4AttributeToExist()
+    })
+
+    it('makes banner visible if view count is less than 3', function () {
+      GOVUK.setCookie('global_banner_seen', JSON.stringify({ count: 1, version: 5 }))
+      globalBanner = new GOVUK.Modules.GlobalBanner(element[0])
+      globalBanner.init()
+
+      expectGlobalBannerToShow()
+      expectGa4AttributeToExist()
+    })
+  })
+})
```

Comment:
Unless I can't see it, should there be a test to ensure the count isn't going above 3 and one that ensures the banner stays hidden if the count is equal to or more than 3?

Reply:
Thanks, have added some new tests.